src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -0,0 +1,234 @@
+package com.openlattice.organizations
+
+import com.codahale.metrics.annotation.Timed
+import com.google.common.base.Preconditions.checkState
+import com.google.common.net.InetAddresses
+import com.openlattice.authorization.*
+import com.openlattice.organization.DatasetApi.Companion.CONTROLLER
+import com.openlattice.organization.DatasetApi
+import com.openlattice.organization.DatasetApi.Companion.COLUMN_NAME
+import com.openlattice.organization.DatasetApi.Companion.COLUMN_NAME_PATH
+import com.openlattice.organization.DatasetApi.Companion.CONNECTION_TYPE
+import com.openlattice.organization.DatasetApi.Companion.CONNECTION_TYPE_PATH
+import com.openlattice.organization.DatasetApi.Companion.DATA
+import com.openlattice.organization.DatasetApi.Companion.EXTERNAL_DATABASE
+import com.openlattice.organization.DatasetApi.Companion.EXTERNAL_DATABASE_COLUMN
+import com.openlattice.organization.DatasetApi.Companion.EXTERNAL_DATABASE_TABLE
+import com.openlattice.organization.DatasetApi.Companion.ID
+import com.openlattice.organization.DatasetApi.Companion.ID_PATH
+import com.openlattice.organization.DatasetApi.Companion.ROW_COUNT
+import com.openlattice.organization.DatasetApi.Companion.ROW_COUNT_PATH
+import com.openlattice.organization.DatasetApi.Companion.TABLE_ID
+import com.openlattice.organization.DatasetApi.Companion.TABLE_ID_PATH
+import com.openlattice.organization.DatasetApi.Companion.TABLE_NAME
+import com.openlattice.organization.DatasetApi.Companion.TABLE_NAME_PATH
+import com.openlattice.organization.DatasetApi.Companion.USER_ID
+import com.openlattice.organization.DatasetApi.Companion.USER_ID_PATH
+import com.openlattice.organization.OrganizationExternalDatabaseColumn
+import com.openlattice.organization.OrganizationExternalDatabaseTable
+import com.openlattice.organization.OrganizationExternalDatabaseTableColumnsPair
+import com.openlattice.postgres.PostgresConnectionType
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import org.springframework.web.bind.annotation.*
+import java.util.*
+import javax.inject.Inject
+
+@RestController
+@RequestMapping(CONTROLLER)
+class DatasetController : DatasetApi, AuthorizingComponent {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(DatasetController::class.java)
+    }
+
+    @Inject
+    private lateinit var edms: ExternalDatabaseManagementService
+
+    @Inject
+    private lateinit var authorizations: AuthorizationManager
+
+    @Inject
+    private lateinit var securableObjectTypes: SecurableObjectResolveTypeService
+
+    @Inject
+    private lateinit var aclKeyReservations: HazelcastAclKeyReservationService
+
+    @Timed
+    @PostMapping(path = [ID_PATH + USER_ID_PATH + CONNECTION_TYPE_PATH + EXTERNAL_DATABASE])
+    override fun addHBARecord(
+            @PathVariable(ID) organizationId: UUID,
+            @PathVariable(USER_ID) userId: String,
+            @PathVariable(CONNECTION_TYPE) connectionType: String,
```

Comment:
You can define it as a `PostgresConnectionType` and then it's easier to check if it's a valid value